src/bp-core/bp-core-filters.php

```diff
@@ -168,7 +168,7 @@ function bp_core_exclude_pages_from_nav_menu_admin( $object = null ) {
  * @return array
  */
 function bp_core_menu_highlight_parent_page( $retval, $page ) {
-	if ( ! is_buddypress() ) {
+	if ( ! is_buddypress() || bp_is_root_blog() ) {
```

Comment:
If we leave this condition this way, it means we don't filter css classes for listed pages if we're not in a BuddyPress area or if it's the WordPress site BuddyPress is activated on. So the `current_page_item` class is never added to the `li` item of the current page when it's actually the case. You can test your patch going on any member's page (eg: site.url/members/anyone) , the class will be missing.

The right check should be, we will not filter the css classes if it's not a BuddyPress area or if we are **not** viewing the WordPress site BuddyPress is activated on. Can you review the PR and add a `!` before `bp_is_root_blog()` to meet this need?

```suggestion
	if ( ! is_buddypress() || ! bp_is_root_blog() ) {
```